src/org/labkey/targetedms/SkylineDocImporter.java

```diff
@@ -233,465 +234,472 @@ private void importSkylineDoc(TargetedMSRun run, File f) throws XMLStreamExcepti
     {
         // TODO - Consider if this is too big to fit in a single transaction. If so, need to blow away all existing
         // data for this run before restarting the import in the case of a retry
+        if (!TargetedMSManager.getSchema().getScope().isTransactionActive())
+        {
+            throw new IllegalStateException("Callers should start their own transaction");
+        }
 
         NetworkDrive.ensureDrive(f.getPath());
         f = extractIfZip(f);
 
-        try (DbScope.Transaction transaction = TargetedMSManager.getSchema().getScope().ensureTransaction();
-             SkylineDocumentParser parser = new SkylineDocumentParser(f, _log, run.getContainer()))
+        TargetedMSService.FolderType folderType = TargetedMSManager.getFolderType(run.getContainer());
+
+        try (SkylineDocumentParser parser = new SkylineDocumentParser(f, _log, run.getContainer()))
         {
-                run.setFormatVersion(parser.getFormatVersion());
-                run.setSoftwareVersion(parser.getSoftwareVersion());
+            run.setFormatVersion(parser.getFormatVersion());
+            run.setSoftwareVersion(parser.getSoftwareVersion());
 
-                ProteinService proteinService = ProteinService.get();
-                ExpData skydData = parser.readSettings(_container, _user);
-                if (skydData != null)
-                {
-                    run.setSkydDataId(skydData.getRowId());
-                }
+            ProteinService proteinService = ProteinService.get();
+            ExpData skydData = parser.readSettings(_container, _user);
+            if (skydData != null)
+            {
+                run.setSkydDataId(skydData.getRowId());
+            }
 
-                // Store the document settings
-                // 0. iRT information
-                run.setiRTscaleId(insertiRTData(parser));
+            // Store the document settings
+            // 0. iRT information
+            run.setiRTscaleId(insertiRTData(parser));
 
+            // 1. Transition settings
+            OptimizationSettings optimizationSettings = insertTransitionSettings(parser.getTransitionSettings());
```

Comment:
Perhaps rename OptimizationSettings to OptimizationInfo to be consistent with ReplicateInfo, ModificationInfo etc.?